neqo-transport/src/connection.rs

```diff
@@ -1173,7 +1191,11 @@ impl Connection {
                         if frame.is_none() {
                             frame = self.send_streams.get_frame(epoch, tx_mode, remaining)
                         }
-                        if frame.is_none() && self.tx_mode == TxMode::Pto {
+
+                        if frame.is_none()
+                            && matches!(self.tx_mode, TxMode::Pto{..})
+                            && self.should_send_ping(epoch)
```

Comment:
Now that we have a whole method to determine whether to send a ping, suggest moving the matches inside it as well.